vllm/core/scheduler.py

```diff
@@ -68,6 +95,73 @@ def lora_requests(self) -> Set[LoRARequest]:
         return {g.lora_request for g in self.scheduled_seq_groups}
 
 
+class SchedulerDecodeOutputs:
```

Comment:
Others can share there thoughts, but I feel that this prefill/decode metadata separation is perhaps not required. If we decide to go ahead with it, I would suggest removing the token_budget ref from these classes -- it is unclear which class has ownership of token budget.

Reply:
I will just have a global token budget which is just an interger and subtract the value based on the output.

Regarding prefill/decoding metadata, I agree with the direction, but I prefer to make this more iterative. So I'd like to suggest to keep the current style and do holistic clean up after e2e chunked prefill PRs are merged. Wdyt?